includes/class-newspack-newsletters.php

```diff
@@ -677,11 +671,18 @@ public static function api_set_mailchimp_interest( $request ) {
 			);
 		}
 
-		if ( 'no_interests' !== $request['interest_id'] && ( ! $field || ! $interest_id ) ) {
-			return new WP_Error(
-				'newspack_newsletters_incorrect_post_type',
-				__( 'Invalid Mailchimp Interest .', 'newspack-newsletters' )
-			);
+		$interest_id_param = $request['interest_id'];
+		if ( $interest_id_param ) {
+			$exploded              = explode( ':', $interest_id_param );
+			$field                 = count( $exploded ) ? $exploded[0] : null;
+			$interest_id           = count( $exploded ) > 1 ? $exploded[1] : null;
+			$is_unsetting_interest = 'no_interests' === $interest_id_param;
+			if ( ! $is_unsetting_interest && ( ! $field || ! $interest_id ) ) {
+				return new WP_Error(
+					'newspack_newsletters_incorrect_post_type',
+					__( 'Invalid Mailchimp Interest .', 'newspack-newsletters' )
```

Comment:
```suggestion
					__( 'Invalid Mailchimp Interest.', 'newspack-newsletters' )
```

Reply:
Fixed in 9067d0f9fd4ff3932b31eca029aedd8d97fb4f0f